### PR TITLE
feat(errors): revamp business logic error codes

### PR DESCRIPTION
**Pull request by g-tejas**
## Problem

Closes FRM-1751

## Solution
`core.errors.ts` contains the error code registry. 

**Breaking Changes** 
- [x] No - this PR is backwards compatible  

## Tests
- [ ] 1. Create a form and keep it private
- [ ] 2. Open datadog dashboard on `staging`
- [ ] 3. Get the public form url and spam 4xxs by opening the website
- [ ] 4. Observe that the traces on datadog are marked with the appropriate error codes. For e.g, Private form errors are 100000 code.

### CODE REVIEW

**Comment by g-tejas** [comment]
* **#7443** <a href="https://app.graphite.dev/github/pr/opengovsg/FormSG/7443?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @g-tejas and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/app/modules/submission/submission.controller.ts [hunk 1]
```diff
@@ -344,7 +346,7 @@ export const streamEncryptedResponses: ControllerHandler<
       meta: logMeta,
       error: cursorResult.error,
     })
-
+    setErrorCode(cursorResult.error)
```
Comment (KenLSM) [inline src/app/modules/submission/submission.controller.ts]: I see that we already have the error codes automatically calling this on error initialisation, is this still required on controllers?

Comment (g-tejas) [reply src/app/modules/submission/submission.controller.ts]: Forgot to leave it out, thanks for the catch 8df12dd

**Review by KenLSM — APPROVED** [review]
LGTM!

**Review by KenLSM — APPROVED** [review]
LGTM!

**Pull request merged**